vividus-plugin-rest-api/src/main/java/org/vividus/http/ExtendedHttpLoggingInterceptor.java

```diff
@@ -0,0 +1,125 @@
+import static org.vividus.http.PublishingAttachmentInterceptor.getMimeType;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.EntityDetails;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.HttpEntityContainer;
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.http.HttpRequestInterceptor;
+import org.apache.hc.core5.http.ProtocolVersion;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.spi.LoggingEventBuilder;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.handler.HttpResponseHandler;
+
+public class ExtendedHttpLoggingInterceptor implements HttpRequestInterceptor, HttpResponseHandler
+{
+    private static final String NEW_LINE = "\n";
+    private static final String HEADERS_FORMAT = "\nHeaders:\n{}";
+    private static final String BODY_FORMAT = "\nBody:\n{}";
+
+    private static final Set<String> LOGGED_CONTENT_TYPES = Set.of(APPLICATION_JSON.getMimeType(),
+            APPLICATION_XML.getMimeType());
+
+    private final Logger logger;
+
+    private final boolean extendedLogging;
+
+    public ExtendedHttpLoggingInterceptor(Class<?> loggerType, boolean extendedLogging)
+    {
+        this.logger = LoggerFactory.getLogger(loggerType);
+        this.extendedLogging = extendedLogging;
+    }
+
+    @Override
+    public void process(HttpRequest request, EntityDetails entityDetails, HttpContext context) throws IOException
+    {
+        String argBrackets = " {}";
+        LoggingEventBuilder loggingEventBuilder = logger.atInfo();
+        StringBuilder loggerFormat = new StringBuilder("Request:");
+
+        ProtocolVersion protocolVersion = request.getVersion();
+        if (protocolVersion != null)
+        {
+            loggingEventBuilder = loggingEventBuilder.addArgument(protocolVersion);
+            loggerFormat.append(argBrackets);
+        }
+        loggingEventBuilder = loggingEventBuilder.addArgument(request);
+        loggerFormat.append(argBrackets);
+
+        if (extendedLogging)
+        {
+            Header[] headers = request.getHeaders();
+            String headersAsString = Stream.of(headers).map(Object::toString).collect(Collectors.joining(NEW_LINE));
+            loggingEventBuilder = loggingEventBuilder.addArgument(headersAsString);
+            loggerFormat.append(HEADERS_FORMAT);
+
+            if (request instanceof HttpEntityContainer httpEntityContainer)
+            {
+                HttpEntity entity = httpEntityContainer.getEntity();
+                if (entity instanceof StringEntity stringEntity)
```

Comment:
if the body is not a string should we add some record like <...binary content> or smth to log? so we can distinguish if the body is empty or not?